src/commands/snapshot.rs

```diff
@@ -1,43 +1,33 @@
 use serenity::{
     framework::standard::{macros::command, Args, CommandResult},
-    model::{channel::Message, id::ChannelId, prelude::MessageId},
+    model::{channel::Message, id::{ChannelId, UserId}, prelude::MessageId},
     prelude::Context
 };
-use std::mem::swap;
-use crate::db::{get_guild};
+use std::{mem::swap, collections::HashMap};
+use crate::commands::{enforce_compliancy, is_moderator};
 #[command] 
 #[min_args(1)]
 #[max_args(2)]
 #[aliases(log_messages, log, snap, snap_messages, snapshot_messages, snip)]
 async fn snapshot(ctx: &Context, msg: &Message, args: Args) -> CommandResult{
     
-    let request = match get_guild(msg.guild_id.unwrap().as_u64()) 
-    {
-        Ok(guild_model) => guild_model,
-        Err(e) => {
-            msg.reply(ctx, format!("Could not get guild from database | {} | This is an error within the code.", e.to_string())).await?;
-            return Ok(());
-        },
-    };
-    
-    if ! request.disclaimer_compliant {
-
-        msg.reply(ctx, format!("A server admin must accept the `{}disclaimer`", dotenv::var("DISCORD_PREFIX").unwrap())).await?;
-        return Ok(());
+    if !is_moderator(ctx, msg).await{
+        msg.reply(ctx, "You must be a moderator to run this command.").await?;
+        return Ok(())
     }
 
-    if ! (match request.mod_role
-    {
-        Some(id) => msg.author.has_role(ctx, msg.guild_id.unwrap(), id).await?,
-        _ => false,
-    } || msg.member(ctx).await.unwrap().permissions(ctx).await.unwrap().administrator())
-    {
-        msg.reply(ctx, "You don't have the required role!").await?;
-        return Ok(());
-    }
+    // get the guild id and cast to i64 so it can be used with the database
+    let guild = *msg.guild_id.unwrap().as_u64();
```

Comment:
i might be missing something - is this guild variable being used any where?

Reply:
yes, rust-analyzer would warn me if it was not

Reply:
oh cool didn't know it did that.  
[nitpick] to call it guild_id instead.  just makes it easier to parse in my opinion.  especially when it's used later.

Reply:
yeah i thought a lot of my variable naming was off and do plan to go back over them so feel free to change them to be nicer whilst you are developing the bot!!! 
@Salivala ^